### PR TITLE
[docs] Add Sentinel info to version-specific upgrade page

### PR DESCRIPTION
**Pull request by angrycub**
The upgrade to sentinel v0.26 is a breaking change for users of custom Sentinel plugins, requiring rebuilding the plugins with sentinel-sdk v4.

### CODE REVIEW

File: website/content/docs/upgrade/upgrade-specific.mdx [hunk 1]
```diff
@@ -13,6 +13,12 @@ upgrade. However, specific versions of Nomad may have more details provided for
 their upgrades as a result of new features or changed behavior. This page is
 used to document those details separately from the standard upgrade flow.
 
+## Nomad 1.9.0 (UNRELEASED)
+
+<EnterpriseAlert inline />
+Nomad Enterprise 1.9.0 includes an updated version of the Sentinel library. Users
+that have built custom Sentinel plugins must recompile them using Sentinel SDK v4.
```
Comment (tgross) [inline website/content/docs/upgrade/upgrade-specific.mdx:20]: Are we waiting till 1.9 to ship the Sentinel upgrade?

**Review by tgross — APPROVED** [review]

**Comment by david-yu** [comment]
@angrycub Is it possible to also include https://developer.hashicorp.com/nomad/docs/release-notes/nomad/v1_8_x#what-s-removed about what has changed from since 1.8.1? 


Also could you clarify what is meant by v4? It seems sentinel-idk is only on v 0.5.x right now: https://github.com/hashicorp/sentinel-sdk/releases

File: website/content/docs/upgrade/upgrade-specific.mdx [hunk 2]
```diff
@@ -13,6 +13,12 @@ upgrade. However, specific versions of Nomad may have more details provided for
 their upgrades as a result of new features or changed behavior. This page is
 used to document those details separately from the standard upgrade flow.
 
+## Nomad 1.8.1 (UNRELEASED)
+
+<EnterpriseAlert inline />
+Nomad Enterprise 1.8.1 includes an updated version of the Sentinel library. Users
+that have built custom Sentinel plugins must recompile them using Sentinel SDK v4.
```
Comment (angrycub) [inline website/content/docs/upgrade/upgrade-specific.mdx:20]:
```suggestion
that have built custom Sentinel plugins must recompile them using Sentinel SDK v0.4 or greater.
```

Comment (angrycub) [inline website/content/docs/upgrade/upgrade-specific.mdx:20]:
```suggestion
<EnterpriseAlert inline />

Nomad Enterprise 1.8.1 includes an updated version of the Sentinel library. Users
that have built custom Sentinel plugins must recompile them using an SDK
supporting Sentinel Plugin Protocol Version 3. Consult the Sentinel
[SDK Compatibility Matrix](https://github.com/hashicorp/sentinel-sdk?tab=readme-ov-file#sdk-compatibility-matrix) for appropriate Sentinel SDK versions.
```

**Pull request merged**